sparta/src/Report.cpp

```diff
@@ -1416,7 +1420,7 @@ void Report::recursAddSubtree_(const TreeNode* n,
             //for(auto& s : stats_){
             //    std::cerr << "  Stat " << s.second->stringize() << " " << s.second->getLocation() << std::endl;
             //}
-            add(n, child_stat_prefix + n->getName());
+            add(n, child_stat_prefix + n->getName(), false);
```

Comment:
`recursAddSubtree_` is already recursively walking the entire tree, so there's no need for `add` to also recursively add its children.

Reply:
Is there a location where the recurse _needs_ to be true?

Reply:
Thanks for this, it made me realize what was really going on:

If you look [here](https://github.com/sparcians/map/pull/497/files#diff-d8d78a9a996d832adee697448c329df4c5cffd5aa39b3cff52afddc051d9705aR1392-R1419) you'll see that we're calling `add()` where the first argument is a `const TreeNode*`. However, we only have a non-const `TreeNode*` overload for `add` [here](https://github.com/sparcians/map/pull/497/files#diff-8e1255d7e3412b851f8a5453e15a77e697369a4bbccab9e5e86e2ce435b67c13L438-R438). Because `StatisticInstance` has a non-explicit constructor that takes a `const TreeNode*`, we actually end up calling the `add` overload for `StatisticInstance&&`.

If you look at the `TreeNode*` overload for `add` [here](https://github.com/sparcians/map/pull/497/files#diff-d8d78a9a996d832adee697448c329df4c5cffd5aa39b3cff52afddc051d9705aL1228-R1244), you'll see that there isn't any recursion happening at all. So I think the original intent was to use this version, but the const-ness was wrong.

My latest commit undoes the extra `recurse` parameter and just changes the `add` signature to take `const TreeNode*`.

Reply:
I tried changing all the single-argument constructors in StatisticInstance.hpp to be `explicit` and everything seems to work fine (regress passes, everything compiles properly). Do you want me to add that to this PR as well to prevent some of this weirdness in the future?

Reply:
Yeah, please do!

Reply:
Done